Source/core/CarbonFootprintCore.js

```diff
@@ -15,14 +15,15 @@ CarbonFootprintCore.prototype.computeFootprint = function(distance) {
 
 CarbonFootprintCore.prototype.footprintToString = function(footprint) {
     var carbonUnit = this.settingsProvider.getCarbonEmissionUnit();
-    if (footprint > 1000 && carbonUnit =='g') {
-        footprint = footprint / 1000;
-        carbonUnit = 'kg';
-    }
-    else if (carbonUnit == 'lbs') {
-        footprint = footprint * 0.00220462;  
-    }
-    footprint = Math.round(footprint);
+  if (footprint < 1 && carbonUnit == 'kg') {
+    footprint = footprint * 1000;
+    carbonUnit = 'g';
+  }
+  else if (footprint < 1 && carbonUnit == 'lbs') {
+    footprint = footprint * 16;
+    carbonUnit = 'oz';
+  }
+    footprint = Math.round(footprint*100)/100;
```

Comment:
But if we have a route with CO2 emission of "300.21 kg", I would rather see "300 kg" instead of "300.21 kg"?

So, if you would like to show more decimal places, do it only for small values of carbon emission, when those decimal places are not negligible.